builtin/game/falling.lua

```diff
@@ -146,13 +146,29 @@ end
 
 local function drop_attached_node(p)
 	local n = core.get_node(p)
+	local node_meta = core.get_meta(p)
+	local meta_fields = node_meta and node_meta:to_table().fields
 	core.remove_node(p)
 	for _, item in pairs(core.get_node_drops(n, "")) do
 		local pos = {
 			x = p.x + math.random()/2 - 0.25,
 			y = p.y + math.random()/2 - 0.25,
 			z = p.z + math.random()/2 - 0.25,
 		}
+		-- If it drops itself, preserve its metadata.
+		if meta_fields then
+			-- item is a table in 0.4.16, but a string in 0.5 - handle both.
+			if (type(item) == "table" and n.name == item.name) or (type(item) == "string" and n.name == item) then
```

Comment:
instead of relying on either of these behaviours you could use
```lua
			if ItemStack(item):get_mame() == n.name then
```

Reply:
That would be less code to maintain, which is good.
But it looks to me like it might run slower.  Is that important here?

Reply:
no, it only runs when metadata is present anyway

Reply:
If you're worried about the performance, you could move the `stack` definition above the if-check and set the count to 1 if the condition applies, which is when `n.name == item.name`.

Reply:
That sounds like a good balance between maintainability and speed.